sdk/core/core-lro/src/lroPoller/azureAsyncPolling.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  failureStates,
+  FinalStateVia,
+  LRO,
+  LROResponse,
+  LROStatus,
+  RawResponse,
+  successStates
+} from "./models";
+
+function getResponseStatus(rawResponse: RawResponse): string {
+  const { status } = rawResponse.body ?? {};
+  return status?.toLowerCase() ?? "succeeded";
+}
+
+function isAzureAsyncPollingDone(rawResponse: RawResponse): boolean {
+  const state = getResponseStatus(rawResponse);
+  if (failureStates.includes(state)) {
+    throw new Error(`Operation status: ${state}`);
+  }
+  return successStates.includes(state);
+}
+
+export function processAzureAsyncOperationResult<TResult>(
+  lro: LRO<TResult>,
+  resourceLocation?: string,
+  finalStateVia?: FinalStateVia
+): (rawResponse: RawResponse, flatResponse: TResult) => LROStatus<TResult> {
+  return (rawResponse: RawResponse, flatResponse: TResult): LROStatus<TResult> => {
+    if (isAzureAsyncPollingDone(rawResponse)) {
+      if (resourceLocation === undefined) {
+        return { rawResponse, flatResponse, done: true };
+      } else {
+        return {
+          rawResponse,
+          flatResponse,
+          done: false,
+          next: async () => {
+            async function sendFinalRequest(): Promise<LROResponse<TResult> | undefined> {
```

Comment:
having this function be declared inside `next` feels a little odd, is it possible to pull it out into module scope and pass in the needed args?

Reply:
Done.